pkg/storage/etcd/etcd_helper.go

```diff
@@ -549,3 +550,12 @@ func (h *etcdHelper) addToCache(index uint64, obj runtime.Object) {
 		metrics.ObserveNewEntry()
 	}
 }
+
+// etcdErrorIndex returns the index associated with the error message and whether the
+// index was available.
+func etcdErrorIndex(err error) (uint64, bool) {
```

Comment:
This is used in 1 place in this file and in 1 place in etcd_watcher.go.
I vote for either inlining it in the code or moving it to etcd/util